presto-main/src/test/java/com/facebook/presto/operator/aggregation/differentialentropy/TestFixedHistogramJacknifeAggregation.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.operator.aggregation.differentialentropy;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class TestFixedHistogramJacknifeAggregation
+        extends TestFixedHistogramAggregation
+{
+    public TestFixedHistogramJacknifeAggregation()
+    {
+        super(Constants.FIXED_HISTOGRAM_JACKNIFE_METHOD_NAME);
+    }
+
+    @Override
+    public Double getExpectedValue(int start, int length)
+    {
+        List<Double> samples = new ArrayList<Double>();
+        List<Double> weights = new ArrayList<Double>();
+        generateSamplesAndWeights(start, length, samples, weights);
+        return calculateEntropy(samples, weights);
+    }
+
+    protected static double calculateEntropy(
```

Comment:
- inline this method
- replace TestFixedHistogramAggregation.calculateEntropy with calculateEntropy
- new ArrayList<Double>(...) -> new ArrayList<>(...)

```
    @Override
    public Double getExpectedValue(int start, int length)
    {
        List<Double> samples = new ArrayList<>();
        List<Double> weights = new ArrayList<>();
        generateSamplesAndWeights(start, length, samples, weights);

        double entropy = samples.size() * calculateEntropy(samples, weights);
        for (int i = 0; i < samples.size(); ++i) {
            List<Double> subSamples = new ArrayList<>(samples);
            subSamples.remove(i);
            List<Double> subWeights = new ArrayList<>(weights);
            subWeights.remove(i);

            double holdOutEntropy = (samples.size() - 1) * calculateEntropy(subSamples, subWeights) / samples.size();
            entropy -= holdOutEntropy;
        }
        return entropy;
    }
```